test/integration/cds_integration_test.cc

```diff
@@ -222,6 +236,168 @@ TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   EXPECT_LE(cx_counter->value(), 1);
 }
 
+// Test that TrafficStats gets created and updated correctly.
+TEST_P(CdsIntegrationTest, TrafficStatsWithClusterCreateUpdateDelete) {
+  this->use_real_stats_ = true;
+
+  initialize();
+  // Ensure that the ClusterInfoImpl is created already.
+  test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
```

Comment:
I mentioned this in another comment but I can't find it now.

These tests are very long and hard to follow. WDYT of factoring out comment blocks so it's easier to see the differences between them?

Reply:
replied

Reply:
Where? I can't see it now (github can make this hard).

I really would like to see stanzas from this test factored out, and for there to be more tests for the variations.